spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/LoadBalancerRestClientHttpServiceGroupConfigurer.java

```diff
@@ -0,0 +1,103 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.client.loadbalancer;
+
+import java.net.URI;
+
+import org.jspecify.annotations.NonNull;
+
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.http.client.service.HttpClientServiceProperties;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.loadbalancer.reactive.ReactiveLoadBalancer;
+import org.springframework.util.function.SingletonSupplier;
+import org.springframework.web.client.RestClient;
+import org.springframework.web.client.support.RestClientHttpServiceGroupConfigurer;
+
+import static org.springframework.cloud.client.loadbalancer.LoadBalancerUriTools.constructInterfaceClientsBaseUrl;
+import static org.springframework.cloud.client.loadbalancer.LoadBalancerUriTools.isServiceIdUrl;
+
+/**
+ * Load-balancer-specific {@link RestClientHttpServiceGroupConfigurer} implementation. If
+ * the group {@code baseUrl} is {@code null}, sets up a {@code baseUrl} with LoadBalancer
+ * {@code serviceId} -resolved from Interface Client {@code groupName} set as
+ * {@code host}. If the group {@code baseUrl} is {@code null} or
+ * {@link LoadBalancerUriTools#isServiceIdUrl(String, String)}, a
+ * {@link DeferringLoadBalancerInterceptor} instance picked from application context is
+ * added to the group's {@link RestClient.Builder} if available, allowing for the requests
+ * to be load-balanced.
+ *
+ * @author Olga Maciaszek-Sharma
+ * @since 5.0.0
+ * @see RestClientHttpServiceGroupConfigurer
+ * @see HttpClientServiceProperties
+ */
+public class LoadBalancerRestClientHttpServiceGroupConfigurer implements RestClientHttpServiceGroupConfigurer {
+
+	// Make sure Boot's configurers run before
+	private static final int ORDER = 10;
+
+	private final ReactiveLoadBalancer.Factory<ServiceInstance> loadBalancerClientFactory;
+
+	private final SingletonSupplier<DeferringLoadBalancerInterceptor> loadBalancerInterceptorSupplier;
+
+	private final HttpClientServiceProperties clientServiceProperties;
+
+	public LoadBalancerRestClientHttpServiceGroupConfigurer(
+			ObjectProvider<DeferringLoadBalancerInterceptor> loadBalancerInterceptorProvider,
+			HttpClientServiceProperties clientServiceProperties,
+			ReactiveLoadBalancer.Factory<ServiceInstance> loadBalancerClientFactory) {
+		this.loadBalancerInterceptorSupplier = SingletonSupplier
+			.ofNullable(loadBalancerInterceptorProvider.getIfAvailable());
+		this.clientServiceProperties = clientServiceProperties;
+		this.loadBalancerClientFactory = loadBalancerClientFactory;
+	}
+
+	@Override
+	public void configureGroups(@NonNull Groups<RestClient.Builder> groups) {
+		DeferringLoadBalancerInterceptor loadBalancerInterceptor = loadBalancerInterceptorSupplier.get();
```

Comment:
In what scenario would `DeferringLoadBalancerInterceptor` not be available?

Reply:
TBH, mainly if the user has manually played with the configuration and botched it.

Reply:
Would it be better then for the auto-configuration to not even run in that case?

Reply:
Honestly, I don't really expect that to happen unless someone completely misconfigures the application, more of a defensive setup, but if you think it's too much, I could just not use the `ofNullable` option to retrieve it and instead expect it to be there.

Reply:
It just seems easier to me (without looking too deeply into it) to not enable the auto-configuration if the classes we need to do so are not present.  At least that is more of the pattern we have followed in the past.